### PR TITLE
Refactor SubtreeSource

### PR DESCRIPTION
**Pull request by edwin0cheng**
This PR simplify `SubtreeSource` by removing `SubtreeWalk` and `Querier` and only walk through the top level `TokenTree` when collecting token from source, by comparing two cursors directly.

### CODE REVIEW

File: crates/ra_mbe/src/subtree_source.rs [hunk 1]
```diff
@@ -1,34 +1,49 @@
...
+pub(crate) struct SubtreeTokenSource<'a> {
+    cached_cursor: Cell<Cursor<'a>>,
     cached: RefCell<Vec<Option<TtToken>>>,
+    curr: (Token, usize),
+}
+
+impl<'a> SubtreeTokenSource<'a> {
+    // Helper function used in test
+    #[allow(unused)]
+    pub fn text(&self) -> SmolStr {
```
Comment (matklad) [inline crates/ra_mbe/src/subtree_source.rs]: maybe `#[cfg(test)]` instead?

File: crates/ra_mbe/src/syntax_bridge.rs [hunk 2]
```diff
@@ -259,64 +258,86 @@ fn convert_tt(
...
-    token_pos: usize,
     inner: SyntaxTreeBuilder,
 
     // Number of roots
     // Use for detect ill-form tree which is not single root
     roots: smallvec::SmallVec<[usize; 1]>,
 }
 
-impl<'a, Q: Querier> TtTreeSink<'a, Q> {
-    fn new(src_querier: &'a Q) -> Self {
```
Comment (matklad) [inline crates/ra_mbe/src/syntax_bridge.rs]: great simplification!

File: crates/ra_mbe/src/subtree_source.rs (hunk 1 above)
Comment (edwin0cheng) [reply crates/ra_mbe/src/subtree_source.rs]: Fixed!

**Pull request merged**